src/Header/Header.js

```diff
@@ -9,7 +9,7 @@ class Header extends Component {
     this.state = {
       search: "",
       searchHidden: true,
-      ref: createRef()
+      searchInput: createRef()
```

Comment:
As someone who isn't on the development team, this renaming to searchInput makes it a lot clearer for me.  Nice work!